fbgemm_gpu/CMakeLists.txt

```diff
@@ -209,15 +285,15 @@ set(cpp_fbgemm_files_avx2 "../src/EmbeddingSpMDMAvx2.cc"
 set_source_files_properties(${cpp_fbgemm_files_avx2}
                             PROPERTIES COMPILE_OPTIONS "-mavx2;-mf16c;-mfma")
 
+set(cpp_fbgemm_files ${cpp_fbgemm_files_normal} ${cpp_fbgemm_files_avx2})
 set(cpp_fbgemm_files_avx512 "../src/EmbeddingSpMDMAvx512.cc")
-
-set_source_files_properties(
-  ${cpp_fbgemm_files_avx512}
-  PROPERTIES COMPILE_OPTIONS
-             "-mavx2;-mf16c;-mfma;-mavx512f;-mavx512bw;-mavx512dq;-mavx512vl")
-
-set(cpp_fbgemm_files ${cpp_fbgemm_files_normal} ${cpp_fbgemm_files_avx2}
-                     ${cpp_fbgemm_files_avx512})
+if(USE_CUDA)
```

Comment:
Is AVX512 related to CUDA?

Reply:
AMD CPU has no support for avx512, and the compiler will complain if adding the compilation flag.

Reply:
In my understanding, HIP is for AMD GPUs so it is irrelevant. For example, I believe we can use AMD GPUs with an Intel Skylake CPU, which has AVX512.

At least, it is not related to CUDA. Can I ask you to fix properly (e.g., detecting the availability of AVX-512 instructions)?

(@jianyuh Please teach me how FBGEMM should handle AVX-512.)